Sources/FoundationEssentials/FileManager/FileManager+Utilities.swift

```diff
@@ -28,8 +28,26 @@ import Glibc
 internal import _CShims
 #elseif os(Windows)
 import CRT
+import WinSDK
 #endif
 
+#if os(Windows)
+package var NSTimeIntervalSince1970: Double {
+    978307200.0
+}
+
+package var NSTimeIntervalSince1601: Double {
+    12622780800.0
+}
+
+extension FILETIME {
+    package var interval: TimeInterval {
```

Comment:
I think it might make more sense to wait to include these as part of a PR that uses them. `NSTimeIntervalSince1970` is already available as `Date.timeIntervalBetween1970AndReferenceDate` so I think we can just use that instead of `NSTimeIntervalSince1970`, and it doesn't look like we use `NSTimeIntervalSince1601` here right?

Depending on the usages, it might make sense to rename this to `timeIntervalSinceReferenceDate` to indicate exactly what this represents and I suspect it can likely just be `internal`, but being able to see where it's used would be quite helpful

Reply:
I can try to pull out parts of the changes that I have that use this. I am using these two constants for converting between `FILETIME` and `TimeInterval`. I don't see any reason to not use `internal` for the second one. However, I am not sure `timeIntervalSinceReferenceDate` is a great name for the constant - perhaps `Date.timeIntervalBetween1601AndReferenceDate` would work?

Reply:
Ah sorry, I was suggesting `timeIntervalSinceReferenceDate` as the property name on `FILETIME` instead of just `interval`, I think having an internal `Date.timeIntervalBetween1601AndReferenceDate` for the constant seems fine if it's needed for Windows

Reply:
Do we want to leave this as just `interval`? What interval does it represent (1601 until the time, or 1970 until the time, or other?)

Reply:
I think so. The `interval` property converts the `FILETIME` to a `TimeInterval`. The `TimeInterval` initialiser uses the UNIX epoch, which is what this converts it to.

Reply:
Gotcha, in that case do you think we should name this `timeIntervalSince1970` instead of just `interval` to make that clearer? My main worry is that `interval` or even `TimeInterval` on its own don't convey the "start" of the interval and it might make this more approachable/readable to specify the start of the interval (or perhaps just outright convert to a `Date` here if we always use this to represent an instance in time).

Also, am I correct in understanding that `FILETIME` represents the time since 1601? If so, shouldn't this be the following:

```swift
Double((self.dwHighDateTime << 32) | self.dwLowDateTime) - Date.timeIntervalBetween1601AndReferenceDate + Date.timeIntervalBetween1970AndReferenceDate
```

I'm not sure subtracting the interval between 1970 and the reference date from the interval between 1601 and the date in question get's us to the right value

Reply:
I suppose that I can name it `timeIntervalSince1970`. It is meant more as a type conversion. I guess the question is, do you think that this is better:

```swift
TimeInterval(ftTime)
```

or is this more obvious:

```
ftTime.interval
```

The calculation itself is correct, the following is from swift-corelibs-foundation:

```swift
extension TimeInterval {
  init(_ ftTime: FILETIME) {
    self = Double((ftTime.dwHighDateTime << 32) | ftTime.dwLowDateTime) - NSTimeIntervalSince1970;
  }
}
```

Reply:
BTW, a FileTime represents the number of 100-nanosecond intervals since the NT Epoch (1/1/1601).

Reply:
Keep in mind `TimeInterval` is simply a typealias of `Double`, so any extensions to it will also appear on any Double as well.